redis/db/redisbp/prometheus.go

```diff
@@ -0,0 +1,128 @@
+package redisbp
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	promNamespace = "redisbp"
+	subsystemPool = "pool"
+
+	nameLabel = "pool"
+)
+
+// exporter provides an interface for Prometheus metrics.
+type exporter struct {
+	client PoolStatser
+	name   string
+
+	poolHitsCounterDesc     *prometheus.Desc
+	poolMissesCounterDesc   *prometheus.Desc
+	poolTimeoutsCounterDesc *prometheus.Desc
+	totalConnectionsDesc    *prometheus.Desc
+	idleConnectionsDesc     *prometheus.Desc
+	staleConnectionsDesc    *prometheus.Desc
```

Comment:
in the case of a service needs more than one redis clients (and thus they will implicitly create more than one `exporter`s), since all those `Desc`s don't really change between the `exporter` instances, I think it's better to just define them as global variables (what you originally did and what I did in the thrift pool one), over making them part of the `exporter` and create during the constructor.